File: backend/src/routes/jobs.js
const express = require('express');
const router = express.Router();

// GET /api/jobs/:jobId — returns current job status for polling
router.get('/:jobId', (req, res) => {
  try {
    const { getJobStatus } = require('../orchestrator/jobManager');
    const status = getJobStatus(req.params.jobId);
    if (!status) return res.status(404).json({ error: 'Job not found' });
    res.json(status);
  } catch (error) {
    console.error('[Jobs] Error fetching job status:', error);
    res.status(500).json({ error: error.message });
  }
});

module.exports = router;

File: backend/app/routers/jobs.py
from fastapi import APIRouter, HTTPException
from app.orchestrator.job_manager import get_job_status

router = APIRouter(prefix="/jobs", tags=["jobs"])

@router.get("/{job_id}")
async def get_job(job_id: str):
    try:
        status = get_job_status(job_id)
        if not status:
            raise HTTPException(status_code=404, detail="Job not found")
        return status
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
